frontend/routes/Wifi.jsx

```diff
@@ -12,42 +29,235 @@ export class Component extends React.Component {
       wifi_networks: [],
       last_refresh: '',
     };
+
     setInterval(() => {
-      exec_table('nmcli dev wifi').then(result => this.setState({ wifi_networks: result }));
-      exec('date').then(result => this.setState({ last_refresh: result }));
+      exec_table('nmcli dev wifi').then(result => {
+        if (!result) return;
+
+        this.setState({
+          wifi_networks: result,
+        });
+      });
+      exec('date').then(result => {
+        this.setState({
+          last_refresh: result,
+        });
+      });
     }, 1000);
   }
 
   render() {
     return (
       <div>
-        <Table data={this.state.wifi_networks} />
+        <EnhancedTable rows={this.state.wifi_networks} />
       </div>
     );
   }
 }
 
-function Table(props) {
-  const { data } = props;
-  if (data.length === 0) return <table />;
+
+// Meta-Data
+
+const in_use = 'IN-USE';
+const first_column = 'IN-USE';
+const active_color = orange[50];
+const headCells = {
+  'IN-USE': { numeric: false, disablePadding: true, label: 'Connected' },
+  BSSID: { numeric: false, disablePadding: false, label: 'BSSID' },
+  SSID: { numeric: false, disablePadding: false, label: 'SSID' },
+  MODE: { numeric: false, disablePadding: false, label: 'Mode' },
+  CHAN: { numeric: true, disablePadding: false, label: 'Channel' },
+  RATE: { numeric: true, disablePadding: false, label: 'Rate' },
+  SIGNAL: { numeric: true, disablePadding: false, label: 'Signal' },
+  BARS: { numeric: false, disablePadding: false, label: 'Bars' },
+  SECURITY: { numeric: false, disablePadding: false, label: 'Security' },
+};
+
+// Meta-Data 
+
+
+function EnhancedTableHead(props) {
+  const { classes, order, orderBy, onRequestSort } = props;
+  const createSortHandler = property => event => {
```

Comment:
inlining this logic would be cleaner IMO

Reply:
Made this inline

Reply:
> Great so far, however
> 
> 1. The formatting is still not completely clean (`npm run format` still reformats some things)
> 2. What do you think about making the security column nicer using chips?
> 3. Finally `Enhanced*` are pretty generic names, I think more meaningful names would be great.
> 
> Also please see the inline comments.

Hi @rroohhh , Please check my new commit. I have made changes as per your suggestions. Also please check in the above suggestions, I have one query for you. 

Regarding recent commit, I have added chips for the security column and it looks good :). I have also checked the dependencies and updated it. Also, I have changed the colour from white to its hex equivalent(#ffff) in index.jsx, since it was causing some problems in rendering.

Do let me know what do you think, meanwhile I will start working on the load graph :)

![Selection_003](https://user-images.githubusercontent.com/51860315/111785027-17e80480-88e2-11eb-9f2d-1545ab224454.png)